runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -13255,6 +13256,452 @@ TR::Register *J9::X86::I386::TreeEvaluator::dwrtbariEvaluator(TR::Node *node, TR
+   dependencies->addPostCondition(firstRegister->getLowOrder(), TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(secondRegister->getLowOrder(), TR::RealRegister::NoReg, cg);
+
+   J9::IA32PrivateLinkage *linkage = static_cast<J9::IA32PrivateLinkage *>(cg->getLinkage(TR_Private));
+   TR::IA32LinkageUtils::pushLongArg(secondChild, cg);
+   TR::IA32LinkageUtils::pushLongArg(firstChild, cg);
+   TR::X86ImmSymInstruction  *instr =
+   generateHelperCallInstruction(node, TR_IA32longRemainder, dependencies, cg);
+   if (!linkage->getProperties().getCallerCleanup())
+      {
+      instr->setAdjustsFramePointerBy(-16);  // 2 long args
+      }
+
+   // Don't preserve eax and edx
+   //
+   instr->setNeedsGCMap(0xFF00FFF6);
+
+   TR::RegisterDependencyConditions  *movDependencies = generateRegisterDependencyConditions((uint8_t)6, 6, cg);
+   movDependencies->addPreCondition(lowRegister, TR::RealRegister::eax, cg);
+   movDependencies->addPreCondition(highRegister, TR::RealRegister::edx, cg);
+   movDependencies->addPostCondition(lowRegister, TR::RealRegister::eax, cg);
+   movDependencies->addPostCondition(highRegister, TR::RealRegister::edx, cg);
+   movDependencies->addPreCondition(firstHigh, TR::RealRegister::NoReg, cg);
+   movDependencies->addPreCondition(secondHigh, TR::RealRegister::NoReg, cg);
+   movDependencies->addPostCondition(firstHigh, TR::RealRegister::NoReg, cg);
+   movDependencies->addPostCondition(secondHigh, TR::RealRegister::NoReg, cg);
+   movDependencies->addPreCondition(firstRegister->getLowOrder(), TR::RealRegister::NoReg, cg);
+   movDependencies->addPreCondition(secondRegister->getLowOrder(), TR::RealRegister::NoReg, cg);
+   movDependencies->addPostCondition(firstRegister->getLowOrder(), TR::RealRegister::NoReg, cg);
+   movDependencies->addPostCondition(secondRegister->getLowOrder(), TR::RealRegister::NoReg, cg);
+
+   generateLabelInstruction(TR::InstOpCode::label, node, doneLabel, movDependencies, cg);
+
+   TR::Register *targetRegister = cg->allocateRegisterPair(lowRegister, highRegister);
+   node->setRegister(targetRegister);
+
+   return targetRegister;
+   }
+
+TR::Register *J9::X86::I386::TreeEvaluator::lstoreEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Compilation *comp = cg->comp();
+   TR::Node *valueChild;
+   TR::SymbolReference *symRef = node->getSymbolReference();
+   bool isVolatile = false;
+
+   if (symRef && !symRef->isUnresolved())
+      {
+      TR::Symbol *symbol = symRef->getSymbol();
+      isVolatile = symbol->isVolatile();
+      TR_OpaqueMethodBlock *caller = node->getOwningMethod();
+      if (isVolatile && caller)
+         {
+         TR_ResolvedMethod *m = comp->fe()->createResolvedMethod(cg->trMemory(), caller, node->getSymbolReference()->getOwningMethod(comp));
+         if (m->getRecognizedMethod() == TR::java_util_concurrent_atomic_AtomicLong_lazySet)
+            {
+            isVolatile = false;
+            }
+         }
+      }
```

Comment:
I think this should be a can be made into a method in OpenJ9, and a stub method that returns `false` in OMR. It can be called something like `lstoreEvaluatorIsVolatileOverride` (or a better name along those lines).